perceval/simulators/loss_simulator.py

```diff
@@ -38,8 +38,11 @@ class LossSimulator(ASimulatorDecorator):
     def _prepare_input(self, input_state):
         return input_state * BasicState([0] * (self._expanded_m - self._original_m))
 
-    def _prepare_circuit(self, circuit):
-        self._original_m = _retrieve_mode_count(circuit)
+    def _prepare_circuit(self, circuit, m = None):
+        if m is None:
+            self._original_m = _retrieve_mode_count(circuit)
+        else:
+            self._original_m = m
```

Comment:
I am unsure, do we need an assert here to verify appropriate m is entered at this stage? Or it won't come to that?